instrumentation/instracer/tracer.go

```diff
@@ -169,17 +169,16 @@ func ParentSpanCtx(ctx context.Context) (jaeger.SpanContext, context.Context) {
 		err     error
 	)
 
+	stringTrace := inslogger.TraceID(ctx)
 	if len(traceSpan.TraceID) > 0 {
-		if len(traceSpan.TraceID) > 32 {
-			traceSpan.TraceID = traceSpan.TraceID[:32]
-		}
-		traceID, err = jaeger.TraceIDFromString(string(traceSpan.TraceID))
-		if err != nil {
-			inslogger.FromContext(ctx).Error(errors.Wrap(err, "failed to parse tracespan traceID"))
-			return emptyContext, ctx
+		stringTrace = string(traceSpan.TraceID)
+	}
+
+	if len(stringTrace) >= 16 {
+		if len(stringTrace) > 32 {
+			stringTrace = stringTrace[:31]
```

Comment:
magic constants everywhere

Reply:
moved to constants and added comment for >= case